streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamSlidingWindowAggregate.java

```diff
@@ -146,13 +142,14 @@ public void processInOrder(final K key, final V value, final long timestamp) {
             boolean leftWinAlreadyCreated = false;
             boolean rightWinAlreadyCreated = false;
 
-            // keep the left type window closest to the record
-            Window latestLeftTypeWindow = null;
+            // Store the previous record
+            Long previousRecord = null;
```

Comment:
Use `previousRecordTimestamp` like in `processEarly`.  You can probably remove the comment then